src/deluge/gui/views/session_view.cpp

```diff
@@ -133,6 +135,44 @@ void SessionView::focusRegained() {
 
 int SessionView::buttonAction(int x, int y, bool on, bool inCardRoutine) {
 
+	if (runtimeFeatureSettings.get(RuntimeFeatureSettingType::MasterCompressor)
+	    == RuntimeFeatureStateToggle::On) { //master compressor
+		int modKnobMode = -1;
+		if (view.activeModControllableModelStack.modControllable) {
+			uint8_t* modKnobModePointer = view.activeModControllableModelStack.modControllable->getModKnobMode();
+			if (modKnobModePointer) modKnobMode = *modKnobModePointer;
+		}
+		if (modKnobMode == 4 && x == modEncoder1ButtonX && y == modEncoder1ButtonY && on) {
+			masterCompEditMode++;
+			masterCompEditMode = masterCompEditMode % 6; //toggle master compressor setting
+
+			if (HAVE_OLED) {
+				modEncoderAction(1, 0);
+			}
+			else {
```

Comment:
How should the Deluge pads and compressor parameters be assigned?
[RELEASE] = Release
[SYNC] = Makeup gain?
[VOL DUCK] = Threshold
[ATTAK] = Attack
[SHAPE] = Ratio
[SEND] = Mix?

Reply:
> How should the Deluge pads and master-compressor parameters be assigned? [RELEASE] = Release [SYNC] = Makeup gain? [VOL DUCK] = Threshold [ATTACK] = Attack [SHAPE] = Ratio [SEND] = Mix?

Sounds fine to me.

Regarding another compressor under the hood: as cool as this would be (having a more efficient comp, even track by track) I would find it really good, if this compressor could be merged before that. It could take a lot of time to change something this profound, and then this really valuable feature isnt available. Maybe having this feature optional via the comm menu could be doable?

Reply:
Yeah I think that makes sense